### PR TITLE
Add groovydoc

### PR DESCRIPTION
**Pull request by Larusso**
## Description
This pull requests adds API docs for all public interfaces and classes. Along with the documentations are slight restructures and some name changes to move this plugin more in line with the other gradle plugins.

## Changes
* ![ADD] groovy doc strings
* ![IMPROVE] restrucure plugin sources

[NEW]:https://atlas-resources.wooga.com/icons/icon_new.svg "New"
[ADD]:http://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:http://resources.atlas.wooga.com/icons/icon_improve.svg "IMPROVE"
[CHANGE]:http://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:http://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:http://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:http://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:http://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:http://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:http://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:http://resources.atlas.wooga.com/icons/icon_webGL.svg "Web:GL"

### CODE REVIEW

File: src/main/groovy/wooga/gradle/github/GithubPlugin.groovy [hunk 1]
```diff
@@ -21,6 +21,37 @@ import org.gradle.api.Plugin
...
+ * Example:
+ * <pre>
+ * {@code
+ *     plugins {
+ *         id "net.wooga.github" version "0.6.1"
+ *     }
+ *
+ *     github {
+ *         username = "wooga"
+ *         password = "a password."
```
Comment (marcolink) [inline src/main/groovy/wooga/gradle/github/GithubPlugin.groovy]: get rid of the dot!?

File: src/main/groovy/wooga/gradle/github/base/GithubBasePlugin.groovy [hunk 2]
```diff
@@ -20,39 +20,42 @@ package wooga.gradle.github.base
...
+     * Value for github gradle task group.
+     * @value "github"
+     */
+    static final String GROUP = "github"
 
     @Override
     void apply(Project project) {
-        this.project = project
-        this.tasks = project.tasks
+        def tasks = project.tasks
```
Comment (marcolink) [inline src/main/groovy/wooga/gradle/github/base/GithubBasePlugin.groovy]: is `tasks` used anywhere?

File: src/main/groovy/wooga/gradle/github/base/tasks/internal/AbstractGithubTask.groovy [hunk 3]
```diff
@@ -141,19 +143,19 @@ abstract class AbstractGithubTask<T extends AbstractGithubTask> extends Conventi
     @Optional
     @Input
     @Override
-    String getUserName() {
-        return this.userName
+    String getUsername() {
+        return this.username
```
Comment (marcolink) [inline src/main/groovy/wooga/gradle/github/base/tasks/internal/AbstractGithubTask.groovy]: do we need `return` ?

**Review by marcolink — APPROVED** [review]

File: src/main/groovy/wooga/gradle/github/base/GithubBasePlugin.groovy (hunk 2 above)
Comment (Larusso) [reply src/main/groovy/wooga/gradle/github/base/GithubBasePlugin.groovy]: Yes. I pass it to this sweet static method.

File: src/main/groovy/wooga/gradle/github/base/tasks/internal/AbstractGithubTask.groovy (hunk 3 above)
Comment (Larusso) [reply src/main/groovy/wooga/gradle/github/base/tasks/internal/AbstractGithubTask.groovy]: No. It is a old habit. But it doesn’t hurt. I could run a quick search and remove pattern over the code.

**Pull request merged**